platform/osgi-extender-fragment/src/main/java/org/motechproject/bundle/extender/MotechOsgiConfigurableApplicationContext.java

```diff
@@ -21,31 +23,35 @@
     private final Object lock = new Object();
     private boolean initialized;
 
+    private static final Logger LOGGER = LoggerFactory.getLogger(MotechOsgiConfigurableApplicationContext.class);
+
+
     /**
      * Constructs the new context using the provided configuration locations.
      * @param configurationLocations the configuration location (Spring xml configuration files)
      */
     public MotechOsgiConfigurableApplicationContext(String[] configurationLocations) {
         super(configurationLocations);
-        addApplicationListener(new ApplicationListener<ApplicationEvent>() {
+        /**addApplicationListener(new ApplicationListener<ApplicationEvent>() {
             @Override
             public void onApplicationEvent(ApplicationEvent event) {
                 if (event instanceof ContextRefreshedEvent) {
+                    LOGGER.debug("Received ContextRefreshedEvent");
                     synchronized (lock) {
                         initialized = true;
                         lock.notifyAll();
                     }
                 }
             }
-        });
+        });*/
```

Comment:
Did you mean to leave this (and `waitForContext` below) commented out? Possibly better to remove if so.